### PR TITLE
Create category docs and organized by category for assets

### PR DESCRIPTION
**Pull request by lykelly19**

### CODE REVIEW

File: scripts/docs/parse_assets.py [hunk 1]
```diff
@@ -31,34 +31,85 @@ def parse_assets(input_dirs: List[Path],
...
-        references[asset_type].append((asset_name, asset_file_name, asset_description))
+        if asset_type not in references:
+            references[asset_type] = defaultdict(list)
+
+        for category in asset_config.categories:
+            references[asset_type][category].append((asset_name, asset_file_name, asset_description))
+
+        # If no categories are found, put the asset under "Uncategorized"
+        if len(asset_config.categories) == 0:
+            references[asset_type]["Uncategorized"].append((asset_name, asset_file_name, asset_description))
```
Comment (elesel) [inline scripts/docs/parse_assets.py:50]: Consider creating constants for "Uncategorized" and "All" (below).

File: scripts/docs/parse_assets.py [hunk 2]
```diff
@@ -31,34 +31,85 @@ def parse_assets(input_dirs: List[Path],
     """
     asset_count = 0
 
-    references = defaultdict(list)
+    references = {}
```
Comment (elesel) [inline scripts/docs/parse_assets.py:34]: You could also use:
```suggestion
    references = defaultdict(lambda: defaultdict(list))
```
See https://stackoverflow.com/a/27809959

File: scripts/docs/parse_assets.py [hunk 3]
```diff
@@ -31,34 +31,85 @@ def parse_assets(input_dirs: List[Path],
...
     for asset_type in references:
+
+        category_docs_links_list = []
+
+        for category in references[asset_type]:
+            if category == "All":
+                continue
+
+            category_doc_file_name = f"{asset_type}s-{category.replace(' ', '-')}-documentation"
+            capitalized_category = category[0].upper() + category[1:]
```
Comment (elesel) [inline scripts/docs/parse_assets.py:65]: https://docs.python.org/3/library/stdtypes.html#str.title
```suggestion
            capitalized_category = category.title()
```

File: scripts/docs/parse_assets.py [hunk 4]
```diff
@@ -31,34 +31,85 @@ def parse_assets(input_dirs: List[Path],
...
+        # If no categories are found, put the asset under "Uncategorized"
+        if len(asset_config.categories) == 0:
+            references[asset_type]["Uncategorized"].append((asset_name, asset_file_name, asset_description))
+
+        references[asset_type]["All"].append((asset_name, asset_file_name, asset_description))
 
     logger.print(f"{asset_count} asset(s) parsed")
 
     for asset_type in references:
+
```
Comment (elesel) [inline scripts/docs/parse_assets.py:57]: No blank line needed here.
```suggestion
```

File: scripts/docs/parse_assets.py [hunk 5]
```diff
@@ -31,34 +31,85 @@ def parse_assets(input_dirs: List[Path],
...
+            with open(f"{asset_type}s/{category_doc_file_name}.md", 'w') as f:
+                f.write(str(category_doc))
+
         # Create a new markdown file for each asset type
         doc = snakemd.new_doc()
-        doc.add_heading(asset_type.capitalize() + "s", level=1)
+        doc.add_heading(f"{asset_type.capitalize()}s", level=1)
+
+        # Add list of asset categories
+        doc.add_heading("Components by categories", level=2)
```
Comment (elesel) [inline scripts/docs/parse_assets.py:93]: "Component" is a specific asset type, so let's avoid that when referring to a assets in general. How about this?
```suggestion
        doc.add_heading("By category", level=2)
```

File: scripts/docs/parse_assets.py (hunk 1 above)
Comment (elesel) [inline scripts/docs/parse_assets.py:50]: Simpler:
```suggestion
        else:
            # Put the asset under "Uncategorized"
            references[asset_type]["Uncategorized"].append((asset_name, asset_file_name, asset_description))
```

File: scripts/docs/parse_assets.py [hunk 6]
```diff
@@ -31,34 +31,85 @@ def parse_assets(input_dirs: List[Path],
...
+
+        references[asset_type]["All"].append((asset_name, asset_file_name, asset_description))
 
     logger.print(f"{asset_count} asset(s) parsed")
 
     for asset_type in references:
+
+        category_docs_links_list = []
+
+        for category in references[asset_type]:
```
Comment (elesel) [inline scripts/docs/parse_assets.py:60]: Should categories be sorted?

File: scripts/docs/parse_assets.py [hunk 7]
```diff
@@ -31,34 +31,85 @@ def parse_assets(input_dirs: List[Path],
...
         doc.add_heading("Glossary", level=2)
 
         doc.add_horizontal_rule()
 
-        for asset_name, asset_file_name, asset_description in references[asset_type]:
+        asset_links_list = []
+        for asset_name, asset_file_name, asset_description in references[asset_type]["All"]:
             doc.add_unordered_list([snakemd.Paragraph(asset_name).insert_link(asset_name, asset_file_name)])
             # limit description to 300 chars
             description = asset_description if len(asset_description) < 300 else (asset_description[:297] + "...")
```
Comment (elesel) [inline scripts/docs/parse_assets.py:108]: The right hand index of slices is exclusive:
```suggestion
            description = asset_description if len(asset_description) <= 300 else (asset_description[:298] + "...")
```

File: scripts/docs/parse_assets.py (hunk 3 above)
Comment (lykelly19) [reply scripts/docs/parse_assets.py:63]: For the category of "Responsible AI", .title() changes it to "Responsible Ai." Should we do some sort of combination or do you have any additional suggestions?

**Pull request closed without merging**